### PR TITLE
[CARBONDATA-2015] Restricted maximum length of bytes per column

### PR DESCRIPTION
**Pull request by dhatchayani**
Validation for number of bytes for a column is added.

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [x] Testing done
        Manual Testing

 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/2839/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/1604/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/2924/

File: processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/NonDictionaryFieldConverterImpl.java [hunk 1]
```diff
@@ -70,13 +70,14 @@ public NonDictionaryFieldConverterImpl(DataField dataField, String nullformat, i
...
+        if (dataType == DataTypes.STRING
+            && value.length > CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT) {
+          throw new CarbonDataLoadingException("Dataload failed, String size cannot exceed "
+              + CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT + " bytes");
         }
         row.update(value, index);
-      } catch (AssertionError ae) {
-        throw new CarbonDataLoadingException("Dataload failed, String size cannot exceed "
-            + CarbonCommonConstants.MAX_CHARS_PER_COLUMN_DEFAULT + " bytes");
+      } catch (CarbonDataLoadingException e) {
```
Comment (manishgupta88) [inline processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/NonDictionaryFieldConverterImpl.java]: Kindly check and confirm if CarbonDataLoadingException can come from any other place also in this flow

Comment (dhatchayani) [reply processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/NonDictionaryFieldConverterImpl.java]: No. There are possibilities of only ParseException and NumberFormatException

**Comment by manishgupta88** [comment]
LGTM

**Comment by shining1989** [comment]
Why Validation for number of bytes for a column is added?

Now I have a column more than 32000 character, insert will failed with error  "String length cannot exceed 32000 characters".

How can i avoid this problem?

**Pull request closed without merging**